src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -271,11 +271,7 @@ protected override void ContextUnloaded(ExecutionContext context)
                     {
                         ExecutionContextState contextState = CurrentContext.GetState<ExecutionContextState>();
                         contextState.NotificationCount += state.NotificationCount;
-                        if (contextState.RequireReturnValue)
-                        {
-                            Push(StackItem.Null);
-                            contextState.RequireReturnValue = false;
-                        }
+                        if (state.PushNullWhenReturn) Push(StackItem.Null);
```

Comment:
Set to false again?

Reply:
It's not needed, the context is unloaded, so `PushNullWhenReturn` won't be used anyway after this.